### PR TITLE
Rubocop upgrade

### PR DESCRIPTION
**Pull request by jamelablack**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -26,16 +25,16 @@ rvm:
...
     - rvm: jruby-18mode
-      env: JRUBY_OPTS='--server -Xcompile.invokedynamic=false'
+    env: JRUBY_OPTS='--server -Xcompile.invokedynamic=false'
     - rvm: jruby
-      env: JRUBY_OPTS='--server -Xcompile.invokedynamic=false'
+    env: JRUBY_OPTS='--server -Xcompile.invokedynamic=false'
   allow_failures:
     - rvm: jruby-head
-      env: JRUBY_OPTS='--server -Xcompile.invokedynamic=false'
+    env: JRUBY_OPTS='--server -Xcompile.invokedynamic=false'
```
Comment (JonRowe) [inline .travis.yml]: Doesn't this change the meaning of this matrix?

File: benchmarks/concatentate_long_strings.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  end
+  x.report('concat with shovel operator') do
+    "something" <<
+    "something"
+  end
+  x.compare!
+end
+
+__END__
+Ruby 2.1.6
```
Comment (JonRowe) [inline benchmarks/concatentate_long_strings.rb]: Great benchmark :)

File: lib/rspec/matchers/built_in/contain_exactly.rb [hunk 3]
```diff
@@ -154,7 +154,7 @@ def unmatched_item_count
             def +(derived_candidate_solution)
               self.class.new(
                 unmatched_expected_indexes + derived_candidate_solution.unmatched_expected_indexes,
-                unmatched_actual_indexes   + derived_candidate_solution.unmatched_actual_indexes,
+                unmatched_actual_indexes + derived_candidate_solution.unmatched_actual_indexes,
```
Comment (JonRowe) [inline lib/rspec/matchers/built_in/contain_exactly.rb]: Personally I'd rather this remain aligned for asthetic reasons, is it possible to disable the cop, what do others think?

Comment (soulcutter) [inline lib/rspec/matchers/built_in/contain_exactly.rb]: I'm all for disabling that cop. Aesthetic whitespace is not a good reason for failing builds IMO - if it's really bad-looking or distractingly inconsistent I think human reviewers are better judges than rubocop.

File: .travis.yml (hunk 1 above)
Comment (soulcutter) [inline .travis.yml]: Yes, I think this is why CI is broken on this PR

**Comment by soulcutter** [comment]
@jamelablack Do you have any interest in continuing on this PR? That benchmark is nice (I would not have guessed that!). I'd be happy to merge if you can wrap it up.

**Comment by samphippen** [comment]
I'm going to close this. @jamelablack if you'd like to continue the PR, let us know, and you can re-open.

**Pull request closed without merging**